src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertiesShouldNotReturnArrays.cs

```diff
@@ -26,14 +26,26 @@ public abstract class PropertiesShouldNotReturnArraysAnalyzer : DiagnosticAnalyz
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/0fss9skc.aspx",
```

Comment:
Since this analyzer applies to both C# and VB, you can make this type sealed and delete the two C# and VB derived type files.